gnomad_qc/v4/sample_qc/sex_inference.py

```diff
@@ -0,0 +1,692 @@
+
+        :param vds: Input VDS to use for sex annotation
+        :param calling_intervals_ht: Calling intervals to filter to for sex annotation
+        :return: Table containing sex annotation for samples in the input VDS
+        """
+        return annotate_sex(
+            vds,
+            included_intervals=calling_intervals_ht,
+            normalization_contig=normalization_contig,
+            sites_ht=freq_ht,
+            aaf_expr="AF",
+            gt_expr="LGT",
+            f_stat_cutoff=f_stat_cutoff,
+            aaf_threshold=aaf_threshold,
+            variants_only_x_ploidy=variants_only_x_ploidy,
+            variants_only_y_ploidy=variants_only_y_ploidy,
+        )
+
+    if platform_ht is not None:
+        logger.info("Collecting platform list...")
+        platforms = platform_ht.aggregate(
+            hl.agg.collect_as_set(platform_ht.qc_platform)
+        )
+
+    if high_cov_intervals or high_cov_by_platform_all:
+        logger.info(
+            "Running sex ploidy and sex karyotype estimation using only high coverage intervals: %d percent of samples "
+            "with greater than %dx coverage on chrX, %d percent of samples with greater than %dx coverage on chrY, and "
+            "%d percent of samples with greater than %dx coverage on %s...",
+            int(prop_samples_x * 100),
+            x_cov,
+            int(prop_samples_y * 100),
+            y_cov,
+            int(prop_samples_norm * 100),
+            norm_cov,
+            normalization_contig,
+        )
+
+        if per_platform or high_cov_by_platform_all:
+            logger.info("Annotating coverage MatrixTable with platform information")
+            coverage_mt = coverage_mt.annotate_cols(
+                platform=platform_ht[coverage_mt.col_key].qc_platform
+            )
+            coverage_mt = coverage_mt.group_cols_by(coverage_mt.platform).aggregate(
+                **{
+                    f"over_{x_cov}x": hl.agg.fraction(coverage_mt.mean_dp > x_cov),
+                    f"over_{y_cov}x": hl.agg.fraction(coverage_mt.mean_dp > y_cov),
+                    f"over_{norm_cov}x": hl.agg.fraction(
+                        coverage_mt.mean_dp > norm_cov
+                    ),
+                }
+            )
+
+        if per_platform:
+            logger.info(
+                "Running sex ploidy and sex karyotype estimation per platform using per platform "
+                "high coverage intervals..."
+            )
+            per_platform_sex_hts = []
+            x_ploidy_cutoffs = {}
+            y_ploidy_cutoffs = {}
+            for platform in platforms:
+                ht = platform_ht.filter(platform_ht.qc_platform == platform)
+                logger.info("Platform %s has %d samples...", platform, ht.count())
+                coverage_platform_mt = _get_filtered_coverage_mt(
+                    coverage_mt.filter_cols(coverage_mt.platform == platform),
+                    hl.agg.any,
```

Comment:
yeah, it is only one value per platform due to the above filter, so I think you are right this can just be `hl.agg.all` like the other, and we can remove `agg_func` from the function above